lib/controllers/share.js

```diff
@@ -67,6 +67,17 @@ function resolveGist(serviceOptions, id) {
         }
     });
 }
+/*
+  Generate short ID by hashing body, converting to base62 then truncating.
+ */
+function shortId(body, length) {
+    console.time('shortid');
+    var base62 = require("base-x")('0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
+    var buf = Buffer.from(body, 'utf-8');
+    var b = base62.encode(require('sha1')(buf));
+    console.timeEnd('shortid');
+    return b.slice(0, length || 16);
```

Comment:
Dunno about "the math", but the full length base62-encoded SHA1 is 54 characters.